### PR TITLE
📝 Update error tracking docs for new flutter-symbols command

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Adds documentation on how to upload files to support deobfuscation and symbolication of Dart error stacks.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests

### CODE REVIEW

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 1]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
 DatadogSdk.instance.initialize(configuration);
 ```
 
-## Manually upload iOS dSYMs to Datadog
-
-Crash reports on iOS are collected in a raw format and mostly contain memory addresses. To map these addresses into legible symbol information, Datadog requires .dSYM files, which are generated in your application's build process.
+If your application suffers a fatal crash, once your application restarts, the Datadog SDK uploads a crash report to Datadog. For non-fatal errors, the Datadog SDK will upload these errors with other RUM data.
```
Comment (alai97) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:30]:
```suggestion
If your application suffers a fatal crash, once your application restarts, the Datadog Flutter SDK uploads a crash report to Datadog. For non-fatal errors, the Datadog Flutter SDK uploads these errors with other RUM data. 
```

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 2]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
...
 
-## Manually upload iOS dSYMs to Datadog
-
-Crash reports on iOS are collected in a raw format and mostly contain memory addresses. To map these addresses into legible symbol information, Datadog requires .dSYM files, which are generated in your application's build process.
+If your application suffers a fatal crash, once your application restarts, the Datadog SDK uploads a crash report to Datadog. For non-fatal errors, the Datadog SDK will upload these errors with other RUM data. 
 
-### Find your dYSM file
 
-Every iOS application produces .dSYM files for each application module. These files minimize an application's binary size and enable faster download speed. Each application version contains a set of .dSYM files. 
+##  Upload Symbol Files to Datadog
```
Comment (alai97) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:33]:
```suggestion
## Upload symbol files to Datadog
```

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 3]
```diff
@@ -2,17 +2,13 @@
...
-</div>
-
 Your crash reports appear in [**Error Tracking**][1].
 
 ## Setup
 
 If you have not set up the Datadog Flutter SDK for RUM yet, follow the [in-app setup instructions][2] or see the [Flutter setup documentation][3].
 
-### Add Crash Reporting
+### Add Native Crash Reporting
```
Comment (alai97) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:11]:
```suggestion
### Add native crash reporting
```

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 4]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
...
-### Find your dYSM file
 
-Every iOS application produces .dSYM files for each application module. These files minimize an application's binary size and enable faster download speed. Each application version contains a set of .dSYM files. 
+##  Upload Symbol Files to Datadog
 
-### Upload your dSYM file
+Native iOS crash reports are collected in a raw format and mostly contain memory addresses. To map these addresses into legible symbol information, Datadog requires that you upload .dSYM files, which are generated in your application's build process.
 
-By uploading your .dSYM file to Datadog, you gain access to the file path and line number of each frame in an error's related stack trace.
+All crash reports that are built with the `--split-debug-info` option set and / or with the `--obfuscate` option set will need to upload their Android Proguard mapping file, and the Dary symbol files generated by the Flutter build process.
```
Comment (alai97) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:37]:
```suggestion
For all crash reports that are built with the `--split-debug-info` option set and/or with the `--obfuscate` option set, you need to upload their Android Proguard mapping file and Dart symbol files generated by the Flutter build process.
```

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 5]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
...
 
-Once your application crashes and you restart the application, the Datadog iOS SDK uploads a crash report to Datadog. 
 
-You can use the [@datadog/datadog-ci][4] command line tool to upload your dSYM file. By default, Flutter adds these dSYM files in `./build/ios/archive/Runner.xcarchive/dSYMs`. After building your application with `flutter build ipa`, run the following shell command to upload your dSYMs to Datadog:
+The [@datadog/datadog-ci][4] command line tool supports uploading all of the necessary files (dSYMs, Android Proguard Mapping, and Dart Symbol Files) in one command.
 
+First, install the `datadog-ci` tool from the instructions above, and create a `datadog-ci.json` file at the root of your project, containing your API key and (optionally) your Datadog site:
```
Comment (alai97) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:42]:
```suggestion
First, install the `datadog-ci` tool from the instructions above and create a `datadog-ci.json` file at the root of your project, containing your API key and (optionally) your Datadog site:
```

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 6]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
...
-}
+Alternately, you can set the `DATADOG_API_KEY` and `DATADOG_SITE` environment variables. 
+
+Then, you can use the following command to upload all the necessary files for symbolication and deobfuscation of your crash reports:
+```sh
+datadog-ci flutter-symbols upload --service-name <your_service_name> --dart-symbols-location <location_of_dart_symbols> --android-mapping --ios-dsyms
 ```
 
-After building your Flutter application with `flutter build apk` or `flutter build appbundle`, use the following shell commands to upload your mapping file to Datadog:
+For a full list of options, refer to the `datadog-ci` [flutter-symbols documentation][6]
```
Comment (alai97) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:59]:
```suggestion
For a full list of options, see the `datadog-ci` [Flutter Symbols documentation][6].
```

**Review by alai97 — CHANGES_REQUESTED** [review]

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 7]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
...
-By uploading your .dSYM file to Datadog, you gain access to the file path and line number of each frame in an error's related stack trace.
+For all crash reports that are built with the `--split-debug-info` option set and/or with the `--obfuscate` option set, you need to upload their Android Proguard mapping file and Dart symbol files generated by the Flutter build process.
 
-Once your application crashes and you restart the application, the Datadog iOS SDK uploads a crash report to Datadog. 
 
-You can use the [@datadog/datadog-ci][4] command line tool to upload your dSYM file. By default, Flutter adds these dSYM files in `./build/ios/archive/Runner.xcarchive/dSYMs`. After building your application with `flutter build ipa`, run the following shell command to upload your dSYMs to Datadog:
+The [@datadog/datadog-ci][4] command line tool supports uploading all of the necessary files (dSYMs, Android Proguard Mapping, and Dart Symbol Files) in one command.
 
+First, install the `datadog-ci` tool from the instructions above and create a `datadog-ci.json` file at the root of your project, containing your API key and (optionally) your Datadog site:
 ```sh
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:43]:
```suggestion
```json
```

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md [hunk 8]
```diff
@@ -31,61 +27,41 @@ final configuration = DdSdkConfiguration(
...
-    // which is set from you pubspec.yaml at build time
-    versionName = "1.3.0" 
-    serviceName = "my-service" // Optional, by default it is read from your Android plugin configuration's package name
-    site = "US" // Optional, can be "US", "EU" or "GOV". Default is "US"
-}
+Alternately, you can set the `DATADOG_API_KEY` and `DATADOG_SITE` environment variables. 
+
+Then, you can use the following command to upload all the necessary files for symbolication and deobfuscation of your crash reports:
+```sh
+datadog-ci flutter-symbols upload --service-name <your_service_name> --dart-symbols-location <location_of_dart_symbols> --android-mapping --ios-dsyms
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/doc/rum/error_tracking.md:56]: why dSyms are mentioned in both `datadog-ci flutter-symbols` call and `npx @datadog/datadog-ci dsyms upload` call?

**Review by xgouchet — APPROVED** [review]

File: packages/datadog_flutter_plugin/doc/rum/error_tracking.md (hunk 8 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/doc/rum/error_tracking.md:56]: Looks like there's some lines tat the end of the readme that shouldn't be there. There shouldn't be any mention of `dsyms upload`

**Review by alai97 — APPROVED** [review]

**Pull request merged**